### PR TITLE
Update misc types and revert WPCompleter export from components

### PR DESCRIPTION
**Pull request by ramonjd**
## What?

Implements feedback from https://github.com/WordPress/gutenberg/pull/67410:

1. Revert WPCompleter export and just update types to `Object` for now. 
2. Use ReactElement instead of ReactNode

### Testing Instructions for Keyboard

Build should pass as expected

### CODE REVIEW

File: packages/block-editor/src/autocompleters/link.js [hunk 1]
```diff
@@ -10,12 +10,10 @@ import { decodeEntities } from '@wordpress/html-entities';
 
 const SHOWN_SUGGESTIONS = 10;
 
-/** @typedef {import('@wordpress/components').WPCompleter} WPCompleter */
```
Comment (ramonjd) [inline packages/block-editor/src/autocompleters/link.js:13]: Removing this as WPCompleter was never exported from the components package until https://github.com/WordPress/gutenberg/pull/67410, where it was advised that it shouldn't be exported.

File: packages/block-editor/src/autocompleters/link.js [hunk 2]
```diff
@@ -10,12 +10,10 @@ import { decodeEntities } from '@wordpress/html-entities';
...
-
 /**
  * Creates a suggestion list for links to posts or pages.
  *
- * @return {WPCompleter} A links completer.
+ * @return {Object} A links completer.
```
Comment (ramonjd) [inline packages/block-editor/src/autocompleters/link.js:16]: Just to have something there. This PR is only meant to avoid type compilation errors for now.

**Review by Mamaduka — APPROVED** [review]
Learning from mistakes 😅 Thanks, @ciampo 🙇

**Comment by ramonjd** [comment]
@Mamaduka that was the fastest ✅ in Github history 😆 

Thank you!!!

**Comment by ciampo** [comment]
Thank you, Ramon!

To be clear, my suggestion was to revert the initial change mostly because we don't want to release new APIs if we're not 100% sure about needing them (since once they're released, it's hard to take them back)>.

A short-term solution for now could be to "duplicate" the type in JSDoc (rather than using Object). And if we actually need that type def, we can consider our options without being rushed by an upcoming plugin/core release :)

**Comment by ramonjd** [comment]
Thanks for closing the loop here @ciampo 

> A short-term solution for now could be to "duplicate" the type in JSDoc (rather than using Object)

I was just sweeping the street with these type changes, to unblock other work but I'll do a quick JSDoc follow up with this suggestion. 👍🏻

**Pull request merged**